xls/modules/zstd/zstd_dec_test.cc

```diff
@@ -1,297 +0,0 @@
-    size_t sim_decomp_size_bytes =
-        (lib_decomp_size + sizeof(uint64_t) - 1) * sizeof(uint64_t);
-    sim_decomp.reserve(sim_decomp_size_bytes);
-
-    // Send compressed frame to decoder simulation
-    for (int i = 0; i < frame.size(); i += 8) {
-      // Pad packet w/ zeros to match the frame size expected by the design.
-      std::array<uint8_t, 8> packet_data = {};
-      auto frame_packet_begin = frame.begin() + i;
-      auto frame_packet_end = frame_packet_begin + 8 < frame.end()
-                                  ? frame_packet_begin + 8
-                                  : frame.end();
-      std::copy(frame_packet_begin, frame_packet_end, packet_data.begin());
-      auto span = absl::MakeSpan(packet_data.data(), 8);
-      auto value = Value(Bits::FromBytes(span, 64));
-      XLS_EXPECT_OK(this->send_queue->Write(value));
-      XLS_EXPECT_OK(this->interpreter->Tick());
-    }
-    PrintTraceMessages("__zstd_dec__ZstdDecoderTest_0_next");
-
-    // Tick decoder simulation until we get expected amount of output data
-    // batches on output channel queue
-    std::optional<int64_t> ticks_timeout = std::nullopt;
-    absl::flat_hash_map<Channel*, int64_t> output_counts = {
-        {this->recv_queue->channel(), sim_decomp_size_words}};
-    XLS_EXPECT_OK(
-        this->interpreter->TickUntilOutput(output_counts, ticks_timeout));
-
-    // Read decompressed data from output channel queue
-    for (int i = 0; i < sim_decomp_size_words; i++) {
-      auto read_value = this->recv_queue->Read();
-      EXPECT_EQ(read_value.has_value(), true);
-      auto packet =
-          ZstdDecodedPacket::MakeZstdDecodedPacket(read_value.value());
-      XLS_EXPECT_OK(packet);
-      auto word_vec = packet->GetData();
-      auto valid_length = packet->GetLength() / CHAR_BIT;
-      std::copy(begin(word_vec), begin(word_vec) + valid_length,
-                back_inserter(sim_decomp));
-    }
-
-    EXPECT_EQ(lib_decomp_size, sim_decomp.size());
-    for (int i = 0; i < lib_decomp_size; i++) {
-      EXPECT_EQ(lib_decomp[i], sim_decomp[i]);
-    }
-  }
-};
-
-/* TESTS */
-
-TEST_F(ZstdDecoderTest, ParseFrameWithRawBlocks) {
-  int seed = 3;  // Arbitrary seed value for small ZSTD frame
-  auto frame = zstd::GenerateFrame(seed, zstd::BlockType::RAW);
-  EXPECT_TRUE(frame.ok());
-  this->ParseAndCompareWithZstd(frame.value());
-}
-
-TEST_F(ZstdDecoderTest, ParseFrameWithRleBlocks) {
-  int seed = 3;  // Arbitrary seed value for small ZSTD frame
-  auto frame = zstd::GenerateFrame(seed, zstd::BlockType::RLE);
-  EXPECT_TRUE(frame.ok());
-  this->ParseAndCompareWithZstd(frame.value());
-}
-
-class ZstdDecoderSeededTest : public ZstdDecoderTest,
-                              public ::testing::WithParamInterface<uint32_t> {
- public:
-  static const uint32_t seed_generator_start = 0;
-  static const uint32_t random_frames_count = 100;
-};
-
-// Test `random_frames_count` instances of randomly generated valid
```

Comment:
curious if that would have done using fuzzing instead?

Reply:
We tried using fuzzing, but the problem is that the control sections need to follow a specific structure and match the properties of the generated data. It might be possible to use fuzzing with some constraints on the frame, but there are a lot of rules to follow, which makes it complicated. In the end, it was easier to use a tool made for creating random test data that meets these requirements.